File: javascript/sieve/sieve.js
var Sieve = function(limit) {
    let nums = [];
    for (let i = 2; i <= limit; i++) {
        nums[i] = true;
    }

    for (let i = 2; i <= limit; i++) {
        if (nums[i]) {
            for (let j = i * 2; j <= limit; j += i) {
                nums[j] = false;
            }
        }
    }

    this.primes = [];
    nums.forEach((value, index) => {
        if (value) { this.primes.push(index); }
    });
}

module.exports = Sieve;

File: ruby/sieve/sieve.rb
class Sieve
    def self.new(num)
        @num = num
        self
    end
    def self.primes
        nums = (2..@num).to_a
        index = 0
        while index < nums.length do
            tmp = nums[index]
            p tmp
            nums.select! { |n| n % tmp != 0 }
            nums.unshift(tmp)
            index += 1
        end
        nums.reverse
    end
end
module BookKeeping
    VERSION = 1
end